File: Python/Syntax/List.py
# List:
#     Taking values as elements of a 1D list:
        ls = list(map(int, input().split()))
        OR, ls = [int(i) for i in input().split()]

    # Taking values as elements of a 2D list:
        ls = [list(map(int, input().split())) for i in range row]

    # Finding out the length of a list or a string:
        len(ls)  or ,  len(s)

    # Summation of all the elements of a list:
        tot = sum(lis)

    # Summation of all the elements of a 2-D list:
        tot = sum(map(sum, ls))

    # Summation of the elements of a list in a certain range:
        tot = sum(ls[0:len(n)])

    # Sorting a list in ascending order:
        ls.sort() or , ls1 = sorted(ls)

    # Sorting a list in descending order:
        ls.sort(reverse=1) or , ls1 = sorted(ls, reverse=1)

    # Assigning all the elements of a 1D list as 0:
        ls = [0] * Number_of_Elements

    # Assigning all the elements of a 2D list as 0:
        ls = [[0] for _ in range(Number_of_Rows)]

    # Maximum value of a list:
        max(ls)

    # Minimum value of a list:
        min(ls)

    # Index of the maximum value of a list:
        ls.index(max(ls))

    # Index of the minimum value of a list:
        ls.index(min(ls))

    # Counting a number or character in a list or string :
        ls.count(element)  or,  s.count(character)

    # Swapping two elements of a list:
        ls[0], ls[1] = ls[1], ls[0]

    # Counting the frequency of an element in a 2-D list :
    #     fre = 0
    #     for i in lis:
    #         print(lis.count(i[0:]))

    # Taking input of a 2-D list:
        r, c = map(int, input().split())
        ls = []
        for i in range(r):
            ls.append(list(map(int, input().split())))

    # Printing space between list elements:
        print(ls[i], end = ' ')

    # Maximum value of a 2-D list:
        max(max(ls, key=max))

    # Finding out the maximum frequent element of a list:
        max(set(ls), key=ls.count)

    # Finding out the frequency of the maximum frequent element of a list:
        ls.count(max(set(ls), key=ls.count))

    # Counting the number of distinct elements of a list:
        len(set(ls))

    # Accessing elements of a list:
        Last element: ls[-1], Second last element: ls[-2]

    # Checking the presence of element in a 2-D list:
        for i in ls:
            if element in i[0:]:
                print("YES")

    # Concatenating two lists:
        ls.extend(ls1)

    # Deleting elements from the list:
        del ls[index]  OR,  del ls[start_index : number_of_elements]
        ls.remove(ele)  # Removes first occurance of the element

    # List of pairs:
        ls = [(0, 0)] * n
        print(ls[0][0], ls[0][1])

    # insert(): inserts an item at the specified index

    # remove(): removes item present at the given index

    # pop(): returns and removes item present at the given index

    # clear(): removes all items from the list

    # index(): returns the index of the first matched item

    # count(): returns the count of the specified item in the list

    # reverse(): reverses the item of the list

    # copy(): returns the shallow copy of the list

File: C#/Syntax/List.cs
// Inlclusions:
    using System.Collections.Generic;

// List
    var ls = new List();
    OR, var ls = new List<int>(n); // List of size n
    ls.Add(x); // Add element x to the end of the list
    ls.RemoveAt(i); // Remove element at index i
    ls.Remove(10); // Remove element 10
    ls.RemoveRange(0, 2); // Remove elements in range [0, 2]
    int x = ls[i]; // Get element at index i
    int n = ls.Count; // Get the size of the list
    ls.Clear(); // Remove all elements from the list
    ls.Sort(); // Sort the list in ascending order
    ls.Reverse(); // Reverse the list
    ls.RemoveAll(x => x == y); // Remove all elements equal to y from the list
    // traverse all
    foreach (int i in ls) {
        Console.WriteLine(i);
    }

// 2-D List:
    var ls = new List<List<int>>(n); // List of size n, where each inner list is also a list
    ls[i].Add(x); // Add element x to the end of the i-th inner list
    ls[i].RemoveAt(j); // Remove element at index j from the i-th inner list
    int x = ls[i][j]; // Get element at index j of the i-th inner list
    int n = ls[i].Count; // Get the size of the i-th inner list
    ls[i].Clear(); // Remove all elements from the i-th inner list
    ls[i].Sort(); // Sort the i-th inner list in ascending order
    ls[i].Reverse(); // Reverse the i-th inner list
    ls[i].RemoveAll(x => x == y); // Remove all elements equal to y from the i-th inner list
    // traverse all
    foreach (var sublist in ls) {
        foreach (var i in sublist) {
            Console.WriteLine(i);
        }
    }

// SortedList:
    var ls = new SortedList();
    OR, var ls = new SortedList<int, string>(); // Sorted list
    ls.Add(1, "one"); // Add element (1, "one") to the sorted list
    ls.Remove(1); // Remove element (1, "one") from the sorted list
    int x = ls.Keys[i]; // Get the key at index i
    string y = ls.Values[i]; // Get the value at index i
    int n = ls.Count; // Get the size of the sorted list
    ls.Contains(key); // Checks whether the list contains key
    OR, ls.ContainsKey(key);
    ls.ContainsValue(val); // Checks whether the list contains value val
    ls.Clear(); // Remove all elements from the sorted list
    // Traversing the elements:
    foreach (var pair in ls) {
        Console.WriteLine("Key: " + pair.Key + ", Value: " + pair.Value);
    }

// LinkedList:
    var ls = new LinkedList<int>();
    OR, var ls = new LinkedList<int>(n); // LinkedList of size n
    ls.AddFirst(x); // Add element x to the beginning of the list
    ls.AddLast(x); // Add element x to the end of the list
    ls.RemoveFirst(); // Remove the first element from the list
    ls.RemoveLast(); // Remove the last element from the list
    ls.Remove(ls.First); // Remove the first element
    ls.Remove(x); // Remove the element x
    int x = ls.First.Value; // Get the first element
    int y = ls.Last.Value; // Get the last element
    int n = ls.Count; // Get the size of the list
    ls.Clear(); // Remove all elements from the list
    ls.Contains(x); // Check availability of x
    // Traversing the elements:
    foreach (var i in ls) {
        Console.WriteLine(i);
    }

// Queue: